snap/snapenv/snapenv.go

```diff
@@ -97,7 +97,7 @@ func userEnv(info *snap.Info, home string) map[string]string {
 	result := map[string]string{
 		"SNAP_USER_COMMON": info.UserCommonDataDir(home),
 		"SNAP_USER_DATA":   info.UserDataDir(home),
-		"XDG_RUNTIME_DIR":  info.UserXdgRuntimeDir(os.Geteuid()),
+		"XDG_RUNTIME_DIR":  info.UserXdgRuntimeDirGlobal(os.Geteuid()),
```

Comment:
It is technically possible that a snap is using evaluating XDG_RUNTIME_DIR and using the old setting, but in practice the things that use /run/user/<uid> are hardcoded to use the directory rather than using environ. mir, gvfs, pulseaudio and wayland all do this and the respective interfaces allow the access. dconf was made to look at environ specifically to support snappy, but it will be ok because the gsettings interface allows access to the updated directory already. I will test various snaps gnome, kde and high profile snaps to make sure they continue to work, but don't foresee issues.

In terms of upgrades, the apparmor policy still allows the old directory and long running applications that happen to evaluate environ (see above) when upgraded will still be able to write there. When they are restarted, they will re-evaluate environ and use the updated directory. /run/user is a volatile directory so the application will simply behave the same as if the system was rebooted (ie, it would be an odd application that stored state in /run/user such that it behaved differently on reboot than on subsequent invocations).